libraries/rib-debug-utils/src/main/java/com/badoo/ribs/core/plugin/utils/debug/AbstractDebugControls.kt

```diff
@@ -0,0 +1,67 @@
+package com.badoo.ribs.core.plugin.utils.debug
+
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import com.badoo.mobile.rib.base.R
+import com.badoo.ribs.core.Rib
+import com.badoo.ribs.core.customisation.inflate
+import com.badoo.ribs.core.plugin.NodeAware
+import com.badoo.ribs.core.plugin.NodeAwareImpl
+import com.badoo.ribs.core.plugin.RibAware
+import com.badoo.ribs.core.plugin.RibAwareImpl
+import com.badoo.ribs.core.plugin.ViewLifecycleAware
+
+@SuppressWarnings("LongParameterList")
+abstract class AbstractDebugControls<T : Rib> internal constructor(
+    private val viewFactory: ((ViewGroup) -> View)? = null,
+    private var viewGroupForChildren: (() -> ViewGroup)? = null,
+    private var growthDirection: GrowthDirection? = null,
+    private val nodeAware: NodeAware = NodeAwareImpl(),
+    private val ribAware: RibAware<T> = RibAwareImpl()
```

Comment:
looks like it is unused? do we need Rib generic type for this class?

Reply:
It's there for children to trigger workflow operations.

See `SwitcherDebugControls` which uses it.